be/src/exec/pipeline/exchange/exchange_sink_operator.cpp

```diff
@@ -296,21 +296,43 @@ ExchangeSinkOperator::ExchangeSinkOperator(OperatorFactory* factory, int32_t id,
     PassThroughChunkBuffer* pass_through_chunk_buffer =
             state->exec_env()->stream_mgr()->get_pass_through_chunk_buffer(state->query_id());
 
-    // fragment_instance_id.lo == -1 indicates that the destination is pseudo for bucket shuffle join.
-    std::optional<std::shared_ptr<Channel>> pseudo_channel;
-
-    for (const auto& destination : destinations) {
+    _channels.reserve(destinations.size());
+    std::vector<int> driver_sequence_per_channel(destinations.size(), 0);
+    for (int i = 0; i < destinations.size(); ++i) {
+        const auto& destination = destinations[i];
         const auto& fragment_instance_id = destination.fragment_instance_id;
-        if (fragment_instance_id.lo == -1 && pseudo_channel.has_value()) {
-            _channels.emplace_back(pseudo_channel.value());
+
+        auto it = _instance_id2channel.find(fragment_instance_id.lo);
+        if (it != _instance_id2channel.end()) {
+            _channels.emplace_back(it->second.get());
         } else {
-            _channels.emplace_back(new Channel(this, destination.brpc_server, fragment_instance_id, dest_node_id,
-                                               _num_shuffles, enable_exchange_pass_through, pass_through_chunk_buffer));
-            if (fragment_instance_id.lo == -1) {
-                pseudo_channel = _channels.back();
+            std::unique_ptr<Channel> channel = std::make_unique<Channel>(
+                    this, destination.brpc_server, fragment_instance_id, dest_node_id, _num_shuffles_per_channel,
+                    enable_exchange_pass_through, pass_through_chunk_buffer);
+            _channels.emplace_back(channel.get());
+            _instance_id2channel.emplace(fragment_instance_id.lo, std::move(channel));
+        }
+
+        if (destination.__isset.pipeline_driver_sequence) {
+            driver_sequence_per_channel[i] = destination.pipeline_driver_sequence;
+            _is_channel_bound_driver_sequence = true;
+        }
+    }
+
+    // Each channel is bound to a specific driver sequence for
+    // the colocate join, local bucket shuffle join, and one-phase aggregate with scan node.
+    if (_is_channel_bound_driver_sequence) {
```

Comment:
it is very hard to understand why the channel in bucket shuffle situtation  only bounds one driver_sequence. Explain why 
 bucketSeq and the driverSeq are introduced?

Reply:
Done.